mongoengine/connection.py

```diff
@@ -385,6 +392,7 @@ def get_db(alias=DEFAULT_CONNECTION_NAME, reconnect=False):
                 conn_settings["password"]
                 or conn_settings["authentication_mechanism"] == "MONGODB-X509"
             )
+            and ("authmechanismproperties" not in conn_settings.keys())
```

Comment:
minor but the .keys() is not needed there
```suggestion
            and ("authmechanismproperties" not in conn_settings)
```